### PR TITLE
Add SSE article for ZFP

### PR DESCRIPTION
**Pull request by markcmiller86**
This is the promised article about software sustainability engineering on ZFP. @lindstro reviewed and approved earlier today though I will leave open the possibility that a few small changes may still be needed before publication on bssw.io.

Be sure to select `main` as the `base` branch above as the target for this PR.

Provide a general summary of your changes in the title above.

Fill in the below **Description** section with minimal text describing the changes/new contributions in this PR and replace `<...>` as appropriate.

Any checklist items that do not apply can be striken out by adding `~~` to the beginning and end of the checklist item as `* ~~[] <checklist-item>~~`.
Also, remove the strikeout markers `~~` for the [wikize_refs.py] checklist items if using formal citations for bssw.io contributions.

# Description

EB Member: @rinkug 

## PR checklist for files displayed on bssw.io site

* [x] ***[Author]*** `@mention` the BSSw.io editorial board member `@<eb-member-id>` in **Description** above assigned to shepherd your PR.
* ~~[ ] ***[Author]*** Add the `<issue-id>` in the **Description** above for the associated GitHub Issue.~~
* ~~[ ] ***[Author]*** Ensure `wikize_refs.py -i <base>.md` is run and commit (if using [wikize_refs.py])~~.
* [x] ***[Author]*** Inspect the content in the `*.md` file(s) as rendered in GitHub for this PR.
* [x] ***[EB Mem]*** Assign this PR to the EB member `<eb-member-id>`.
* [x] ***[EB Mem]*** Assign this PR to the author of the PR `<pr-author-id>`.
* [x] ***[EB Mem]*** Add one or more Reviewers.
* [x] ***[EB Mem]*** Add label `content: <content-type>` for the type of contribution.
* [x] ***[EB Mem]*** Add to Project `Content Development` (see [Content Development]).
* [x] ***[EB Mem]*** Add [meta-data] to the `*.md` file(s) (set `Publish: yes`).
* [x] ***[EB Mem]*** Add label `preview` (so PR branch will be merged to 'preview' branch and watch for possible merge failures).
* [x] ***[EB Mem]*** Rebuild [preview] site and confirm new content is there, renders correctly and is returned in searches.
* [x] ***[Author]*** Make any final changes to the PR based on feedback.
* ~~[ ] ***[Author]*** Ensure `wikize_refs.py -i <base>.md` is run and commit (if using [wikize_refs.py]).~~
* [x] ***[EB Mem]*** Rebuild [preview] site and re-confirm content looks correct.
* [x] ***[EB Mem]*** Ensure at least one reviewer signs off on the final changes.
* [x] ***[EB Mem]*** Move the PR to "Ready to Publish" in [Content Development].
* [x] ***[EB Mem]*** Leave comment and @mention `@betterscientificsoftware/bssw-maint` (**BSSw Maint**) asking to carry out final publication steps.
* [ ] ***[BSSw Maint]*** Verify that all needed files are present in the PR (article, images, updates to Site/Homepage.md carousel and/or Site/Announcements/Announcements.md as appropriate).
* [ ] ***[BSSw Maint]*** Merge PR. (Should automatically move to "Done" in [Content Development].)
* [ ] ***[BSSw Maint]*** Verify new contribution shows up on [bssw.io] as expected.

NOTE:
* Checklist items prefixed with ***[Author]*** are expected to be performed by the author of the PR or can be performed by the author.
* Checklist items prefixed with ***[EB Mem]*** must be performed by a [BSSw.io Editorial Board (EB) Member](https://betterscientificsoftware.github.io/bssw.io/bssw_members.html).
* Checklist items prefixed with ***[BSSw Maint]*** must be performed by someone in the `@betterscientificsoftware/bssw-maint` team (hint: type `@`, `b`, `s`, `s`, `w`, `-`, `m`  to auto-complete to `@betterscientificsoftware/bssw-maint`).

[preview]: https://preview.bssw.io
[bssw.io]: https://bssw.io
[Content Development]: https://github.com/betterscientificsoftware/bssw.io/projects/3
[BSSw Internal]: https://github.com/betterscientificsoftware/bssw.io/projects/2
[meta-data]: https://betterscientificsoftware.github.io/bssw.io/bssw_styling_common.html#metadata-section
[wikize_refs.py]: https://github.com/betterscientificsoftware/bssw.io/blob/master/utils/README.md#wikize_refspy

### CODE REVIEW

**Review by bartlettroscoe — APPROVED** [review]
I read the article is it reads well.  I think this is a good addition to BSSw.io.  It would make a good featured blog article.

**Comment by markcmiller86** [comment]
@bartlettroscoe thanks for the positive feedback 🎉 . That said, I am sheepish about merging myself because prior to ECP end, I would defer to final decision from @rinkug or maybe @bernhold or @curfman. Should I be wearinig a new thinking cap now?

**Comment by bartlettroscoe** [comment]
>  That said, I am sheepish about merging myself because prior to ECP end, I would defer to final decision from @rinkug or maybe @bernhold or @curfman.

@markcmiller86, that is a good question.  But until we define the updated process, we should not assume that a positive review by a BSSw.io editorial member is sufficient for merging an article.  I think we should follow the checklist process above.

**Comment by markcmiller86** [comment]
@betterscientificsoftware/bssw-maint this is ready for publication

**Comment by rinkug** [comment]
@bernhold : We have another short article ready. @bartlettroscoe has reviewed it. Perhaps you can also look at it and merge it and put it on the digest highlight schedule?

**Comment by bernhold** [comment]
Hi @markcmiller86, I'm editing this for publication later this week.  I've taken the liberty of restructuring it to be more in the format of an interview.  All I really did was to intersperse your questions into appropriate places in Peter's answers.  And I moved one of Peter's paragraphs from one section (question) to another where it made more sense.  Can you please give this version a quick scan and confirm that I haven't messed things up (too much)?  Thanks

**Comment by markcmiller86** [comment]
@bernhold LGTM. Thanks

File: Articles/Blog/2024-01-zfp-sse.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+I used to maintain zfp release assets in a somewhat adhoc fashion on the LLNL Computing department's web server.
+[Ian Lee](https://github.com/IanLee1521) took it upon himself to set up a GitHub repo for zfp (and other products), which really got the ball rolling on starting new SSE practices.
+
+**Mark:** If you had it to do all over again (I mean the SSE learning curve/ramp), how would you do it differently and/or what resources would have made it easier/faster?
+
+**Peter:** To be honest, I don't think I would have benefited as much from taking formal training to learn the tools and practices I now use daily.
+I feel that I need to get my hands dirty and *do* in order to learn.
+But it was very helpful to work with and learn from someone (Markus Salasoo, a zfp developer who had industry experience with SSE).
+I think I relied a bit too much on Markus in the beginning.
+It was only after he left the project that I really had to role up my own sleeves and get immersed in git/GitHub and learn how to use these and other tools.
```
Comment (lindstro) [inline Articles/Blog/2024-01-zfp-sse.md:37]: role up -> roll up

File: Articles/Blog/2024-01-zfp-sse.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+With the amount of documentation we now have, it would have been impossible to scale things up with a less capable documentation toolchain.
+
+**Mark:** Humans are often bad at judging cost/benefit trade offs.
+When the [VisIt](https://visit.llnl.gov) project [transitioned from Subversion to git/GitHub](https://bssw.io/blog_posts/continuous-technology-refreshment-an-introduction-using-recent-tech-refresh-experiences-on-visit), we were all worried about the added time and effort needed to review each other's pull requests.
+Within just a week or two of being on GitHub, we had caught enough problems in pull request reviews that we were quickly convinced the cost was worth the benefit.
+
+**Mark:** Where do you think SSE practices on zfp compare with other similar projects you are aware of or familiar with?
+
+**Peter:** When it comes to SSE practices (e.g. testing, documentation, code quality, and portability), I think zfp is in a good place when compared with other, similar packages.
+Developers working on similar packages have even commented that zfp works out of the box without hitches while other packages do not fair so well -- something even documented in published papers.
```
Comment (lindstro) [inline Articles/Blog/2024-01-zfp-sse.md:61]: fair -> fare

**Review by lindstro — CHANGES_REQUESTED** [review]
Looks great.  I have only a couple of minor corrections.

File: Articles/Blog/2024-01-zfp-sse.md (hunk 1 above)
Comment (markcmiller86) [reply Articles/Blog/2024-01-zfp-sse.md]: Fixed in c9cda8b8a4678cc5efd79a19c8e75c13a58dfccd

File: Articles/Blog/2024-01-zfp-sse.md (hunk 2 above)
Comment (markcmiller86) [reply Articles/Blog/2024-01-zfp-sse.md]: Fixed in c9cda8b8a4678cc5efd79a19c8e75c13a58dfccd

**Pull request merged**